### PR TITLE
Route created to get all company goals

### PR DESCRIPTION
**Pull request by Ridwan0029**

### CODE REVIEW

File: controllers/goalController.js [hunk 1]
```diff
@@ -3,6 +3,22 @@ const Joi = require('joi');
...
+
+//   res.status(200).json({ status: 'success', data: [{ foo: 'bar' }] });
+// });
+
+exports.getAllGoals = catchAsync(async (req, res, next) => {
+  const goals = await axios.get(`https://test-zuri-core.herokuapp.com/crud/goals/find`);
+  //console.log(goals);
+  
+  // Sending Responses
+  res.status(200).json({ status: 'success', goals: { ...goals.data.data } });
```
Comment (Asylcreek) [inline controllers/goalController.js:19]: Please change 'goals:' to 'data:'

File: controllers/goalController.js [hunk 2]
```diff
@@ -3,6 +3,22 @@ const Joi = require('joi');
 
 const catchAsync = require('../utils/catchAsync');
 
+// exports.getAllGoals = catchAsync(async (req, res, next) => {
+//   // If this promise is rejected, catchAsync would catch it
+//   // and send it to the globalErrorHandler
+//   await Promise.resolve('hi');
+
+//   res.status(200).json({ status: 'success', data: [{ foo: 'bar' }] });
+// });
```
Comment (Asylcreek) [inline controllers/goalController.js:12]: These lines can be deleted

File: package.json [hunk 3]
```diff
@@ -1,4 +1,61 @@
 {
+<<<<<<< HEAD
```
Comment (Asylcreek) [inline package.json:2]: Finish the merge of this file. Be sure to pull from staging again before you push

File: routes/goalRoutes.js [hunk 4]
```diff
@@ -1,5 +1,10 @@
 const { Router } = require('express');
 
+const {getAllGoals, createGoal} = require('../controllers/goalController');
+
+const router = Router();
+
+router.route('/').get(getAllGoals).post(createGoal);
```
Comment (Asylcreek) [inline routes/goalRoutes.js:7]: Check this file again, you do not need some imports, and declarations. They already exist.

Make sure every other person's code works too

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: controllers/goalController.js (hunk 1 above)
Comment (Ridwan0029) [reply controllers/goalController.js]: done

File: controllers/goalController.js (hunk 2 above)
Comment (Ridwan0029) [reply controllers/goalController.js]: done

File: package.json (hunk 3 above)
Comment (Ridwan0029) [reply package.json]: done

File: routes/goalRoutes.js (hunk 4 above)
Comment (Ridwan0029) [reply routes/goalRoutes.js]: done

**Review by Asylcreek — CHANGES_REQUESTED** [review]
Delete your yarn-error.log and add it to your gitignore

File: routes/goalRoutes.js [hunk 5]
```diff
@@ -1,15 +1,10 @@
...
-const router = Router();
-
-router.route('/').get(getAllGoals).post(createGoal);
 const { createGoals, getSingleGoal, getAllGoals } = require('../controllers/goalController');
 
 const router = Router();
 
-
+router.route('/').get(getAllGoals).post(createGoals);
 router.route('/').post(createGoals).get(getAllGoals);
```
[the remark this answers is not in the document]
Comment (Ridwan0029) [reply routes/goalRoutes.js:8]: How?  One is get, the other is post

[the remark this answers is not in the document]
Comment (Asylcreek) [reply routes/goalRoutes.js:8]: Did you read the lines at all?

It has the same implementations, just in a different order

**Comment by Asylcreek** [comment]
Please delete the yarn-error.log and add it to gitignore

**Pull request closed without merging**